var/spack/repos/builtin/packages/lfortran/package.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Lfortran(CMakePackage):
+    """Modern interactive LLVM-based Fortran compiler"""
+
+    homepage = "https://lfortran.org"
+    url = "https://lfortran.github.io/tarballs/release/lfortran-0.19.0.tar.gz"
+    git = "https://github.com/lfortran/lfortran.git"
+    maintainers = ["certik"]
+
+    # The build process uses 'git describe --tags' to get the package version
+    version("main", branch="main", get_full_repo=True)
+    version("0.19.0", sha256="d496f61d7133b624deb3562677c0cbf98e747262babd4ac010dbd3ab4303d805")
+
+    variant("llvm", default=True, description="Build with LLVM support")
+    variant("stacktrace", default=True, description="Build with stacktrace support")
+
+    depends_on("python@3:", type="build", when="@main")
+    depends_on("cmake", type="build")
+    depends_on("llvm@14.0.4", type="build", when="+llvm")
+    depends_on("zlib@1.2.12", type="build")
+    depends_on("re2c@2.2", type="build", when="@main")
+    depends_on("bison@3.4.2", type="build", when="@main")
+    depends_on("binutils@2.38", type="build", when="platform=linux")
+
+    @run_before("cmake")
+    def prepare_source(self):
+        if self.spec.satisfies("@main"):
```

Comment:
@adamjstewart @becker33 is there a way to have two @ attributes? i.e. `@run_before("cmake")` and `@when("@main")` ?

Reply:
@alalazo would be the expert here. I would try it and try both orders. It is sensitive to order.

Reply:
I did, but it didn't work! @alalazo any ideas?

Reply:
Ok, I just learned that `LFORTRAN_BUILD_ALL=no` runs `build0.sh` as part of CMake, so we can simplify this even more! @certik chime in if I am wrong here.

Reply:
I think this is correct.